src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -136,35 +137,62 @@ public static class TarUtils
                 var output = new List<string>();
                 Task readLines(string prefix, StreamReader reader) => Task.Run(async () =>
                 {
-                    string line;
-                    while (null != (line = await reader.ReadLineAsync()))
+                    try
                     {
-                        lock (output)
+                        string line;
+                        while (null != (line = await reader.ReadLineAsync()))
                         {
-                            output.Add($"{prefix}{line}");
+                            lock (output)
+                            {
+                                output.Add($"{prefix}{line}");
+                            }
                         }
                     }
+                    catch (Exception e)
+                    {
+                        context.Error(e.ToString());
+                    }
                 });
+
                 Task readStdOut = readLines("stdout: ", process.StandardOutput);
                 Task readStdError = readLines("stderr: ", process.StandardError);
 
                 // Our goal is to always have the process ended or killed by the time we exit the function.
                 try
                 {
-                    using (cancellationToken.Register(() => process.Kill()))
+                    //using (cancellationToken.Register(() => process.Kill()))
```

Comment:
if we no longer need this line, can we remove it and also the now unnecessary scoping brackets?

Reply:
Still in testing.